common/include/pcl/point_types_conversion.h

```diff
@@ -134,6 +136,59 @@ namespace pcl
     if (out.h < 0.f) out.h += 360.f;
   }
 
+  /** \brief Convert a XYZRGB-based point type to a XYZLAB
+    * \param[in] in the input XYZRGB(XYZRGBA, XYZRGBL, etc.) point
+    * \param[out] out the output XYZLAB point
+    */
+  template <typename PointT, traits::HasColor<PointT> = true>
+  inline void
+  PointXYZRGBtoXYZLAB (const PointT& in,
+                        PointXYZLAB&        out)
+  {
+    out.x = in.x;
+    out.y = in.y;
+    out.z = in.z;
+    out.data[3] = 1.0; // important for homogeneous coordinates
+
+    // convert sRGB to CIELAB
+    const Eigen::Vector3i colorRGB = in.getRGBVector3i();
+
+    // for sRGB   -> CIEXYZ see http://www.easyrgb.com/index.php?X=MATH&H=02#text2
+    // for CIEXYZ -> CIELAB see http://www.easyrgb.com/index.php?X=MATH&H=07#text7
+    // an overview at: https://www.comp.nus.edu.sg/~leowwk/papers/colordiff.pdf
+
+    const auto& sRGB_LUT = RGB2sRGB_LUT<double, 8>();
+
+    const double R = sRGB_LUT[colorRGB[0]];
+    const double G = sRGB_LUT[colorRGB[1]];
+    const double B = sRGB_LUT[colorRGB[2]];
```

Comment:
Couldn't you use `in.r`, `in.g`, and `in.b` directly? And delete `colorRGB`?

Reply:
Thanks for your advice, I'll fix it then. :grinning: